spicy/toolchain/src/ast/types/unit.cc

```diff
@@ -14,3 +16,35 @@ std::optional<type::unit::Item> type::Unit::field(const ID& id) const {
 
     return {};
 }
+
+struct AssignIndicesVisitor : public hilti::visitor::PreOrder<void, AssignIndicesVisitor> {
```

Comment:
Can we go with e.g., `AssignFieldIndicesVisitor` instead to make clear these are indices of fields?